common/src/main/java/com/wynntils/mc/render/FontRenderer.java

```diff
@@ -133,12 +133,40 @@ public void renderText(
         List<String> parts = Arrays.stream(StringUtils.wrapTextBySize(text, (int) maxWidth))
                 .filter(s -> !s.isBlank())
                 .toList();
+        String lastPart = "";
         for (int i = 0; i < parts.size(); i++) {
-            String part = parts.get(i);
+            // copy the format codes to this part as well
+            String part = getLastPartCodes(lastPart) + parts.get(i);
+            lastPart = part;
             renderText(poseStack, part, x, y + (i * NEWLINE_OFFSET), customColor, alignment, shadow);
         }
     }
 
+    private String getLastPartCodes(String lastPart) {
+        if (!lastPart.contains("§")) return "";
+
+        String lastPartCodes = "";
+        int index;
+        while ((index = lastPart.lastIndexOf("§")) != -1) {
+            if (index >= lastPart.length() - 1) {
+                // trailing §, no format code, skip it
+                lastPart = lastPart.substring(0, index);
+                continue;
+            }
+            String thisCode = lastPart.substring(index, index + 2);
+            if (thisCode.charAt(1) == 'r') {
+                // it's a reset code, we can stop looking
+                break;
+            }
+            // prepend to codes since we're going backwards
+            lastPartCodes = thisCode + lastPartCodes;
+
```

Comment:
You know what, Qodana? I'm starting to get really pissed off with you. There is no `StringBuild.prepend()`. This rule of yours should have checked the order of the arguments to the `+`, not just naively assume that we don't know how to use a StringBuilder.